### ISSUE TITLE
masked.softmax/masked.softmin OpInfo samples can be fully masked-out, making test_vmapvjpvjp compare undefined values

### ISSUE THREAD

**Issue by tszulist-hbn** [pytorch/pytorch#196289]
### 🐛 Describe the bug

`test/functorch/test_ops.py::TestOperatorsDevice::test_vmapvjpvjp_masked_softmax_*_float32` and `..._masked_softmin_*_float32` fail with a nan-vs-0 mismatch, depending on the RNG stream of the device under test.

The failing sample is the 0-d case from `sample_inputs_masked_softmax`: a scalar input, `dim=0`, and a mask drawn by `_generate_masked_op_mask` via `make_tensor(input_shape, dtype=torch.bool, ...)`. For a 0-d input that mask is a single random bool. When it comes out `False` the input is *fully masked out*, which `torch/masked/_ops.py` explicitly documents as undefined:

> If all elements of `input` along the given `dim` are ignored (fully masked-out), the corresponding element of the output tensor will have undefined value

The eager path yields `nan` and the vmap path yields `0`, so `assertEqual` fails on a value the op does not promise anything about.

Repro (no test-suite imports):

```python
import torch
from torch.func import vjp

mask = torch.tensor(False)
x = torch.tensor(-8.478373527526855, requires_grad=True)
cot = torch.tensor(0.5046969056129456, requires_grad=True)

out = torch.masked.softmax(x, 0, mask=mask)
(gx,) = torch.autograd.grad(out, x, cot, create_graph=True)
(g_cot,) = torch.autograd.grad(gx, cot, torch.ones_like(gx))
print(out.item(), gx.item(), g_cot.item())  # nan 0.0 nan

def inner(t, c):
    _, vjp_fn = vjp(lambda u: torch.masked.softmax(u, 0, mask=mask), t)
    return vjp_fn(c)[0]

fg, fvjp = vjp(inner, x, cot)
print(fvjp(torch.ones_like(fg))[1].item())  # nan; vmap of the same gives 0.0
```

`masked.softmin` behaves identically. Both reproduce on CPU and XPU.

### Why it surfaces per-device

The per-sample RNG in `TrackedInputIter.__next__` is seeded from `sha256(test_fn.__qualname__)`, so which bool the mask draws depends on both the test's qualified name and the device's RNG stream. CPU has carried an `xfail` for `masked.softmax` here for a long time; https://github.com/pytorch/pytorch/pull/193412 renames the test class, which reshuffles the draw and makes CUDA (13.0 and 13.2), ROCm mi350 and XPU hit the fully-masked mask too.

### Suggested fix

The cleanest fix is in the sample generator rather than in the skip lists: `_generate_masked_op_mask` should not emit a fully masked-out mask for ops whose fully-masked output is documented undefined (or the OpInfo should mark such samples so gradient tests skip them). That would let all the `masked.softmax`/`masked.softmin` entries in `test_vmapvjpvjp`'s `skipOps` go away, on every device.

### Versions

<details>
<summary><code>python -m torch.utils.collect_env</code></summary>

```
Collecting environment information...
PyTorch version: 2.15.0a0+gita4cd817
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.4 LTS (x86_64)
GCC version: (Ubuntu 14.2.0-4ubuntu2~24.04.1) 14.2.0
Clang version: Could not collect
CMake version: version 3.31.6
Libc version: glibc-2.39

Python version: 3.12.3 (main, Jun 19 2026, 12:46:00) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.17.0-35-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: True
XPU used to build PyTorch: 20260100
Intel GPU driver version:
* libze1:	1.24.0.0-1146~24.04
* intel-opencl-icd:	25.18.33578.77-1146~24.04
Intel GPU models onboard:
* Intel(R) Data Center GPU Max 1550
Intel GPU models detected:
* [0] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD5, uuid=8680d50b-2f00-0000-8c00-000000000001, driver_version='1.6.33578+77', total_memory=65520MB, local_mem_size=128KB, last_level_cache_size=196608KB, max_compute_units=512, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=512, gpu_subslice_count=64, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [1] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD5, uuid=8680d50b-2f00-0000-8c00-000000000002, driver_version='1.6.33578+77', total_memory=65520MB, local_mem_size=128KB, last_level_cache_size=196608KB, max_compute_units=512, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=512, gpu_subslice_count=64, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           46 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  64
On-line CPU(s) list:                     0-63
Vendor ID:                               GenuineIntel
Model name:                              Intel(R) Xeon(R) Platinum 8352Y CPU @ 2.20GHz
CPU family:                              6
Model:                                   106
Thread(s) per core:                      2
Core(s) per socket:                      32
Socket(s):                               1
Stepping:                                6
CPU(s) scaling MHz:                      25%
CPU max MHz:                             3400.0000
CPU min MHz:                             800.0000
BogoMIPS:                                4400.00
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities
Virtualization:                          VT-x
L1d cache:                               1.5 MiB (32 instances)
L1i cache:                               1 MiB (32 instances)
L2 cache:                                40 MiB (32 instances)
L3 cache:                                48 MiB (1 instance)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-63
Vulnerability Gather data sampling:      Vulnerable
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Mitigation; Aligned branch/return thunks
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Old microcode:             Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Not affected

Versions of relevant libraries:
[pip3] intel-cmplr-lib-ur==2026.0.0
[pip3] intel-openmp==2026.0.0
[pip3] mkl-include==2024.2.0
[pip3] mkl-static==2024.2.0
[pip3] mypy==1.16.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.26.4
[pip3] nvidia-cuda-cupti==13.3.75
[pip3] onemkl-license==2026.0.0
[pip3] onnx==1.21.0
[pip3] onnx-ir==0.1.16
[pip3] onnxscript==0.6.2
[pip3] optree==0.13.0
[pip3] tbb==2021.13.1
[pip3] tbb-devel==2023.0.0
[pip3] tcmlib==1.5.0
[pip3] torch==2.15.0a0+gita4cd817
[pip3] triton-xpu==3.7.1+git21033c4e
[pip3] umf==1.1.0
[conda] Could not collect
```

</details>

cc @mruberry

**Comment by bodapatisaikrishna** [comment]
Opened #196300. `sample_inputs_masked_softmax` now skips any mask that leaves a slice fully masked out along the softmax `dim` (covers softmax/softmin/log_softmax), and the CPU `xfail` on `test_vmapvjpvjp_masked_softmax` is removed. Non-degenerate samples are unchanged.
